packages/realm/src/app-services/SyncSession.ts

```diff
@@ -273,6 +290,7 @@ export class SyncSession {
   /** @internal */
   constructor(internal: binding.SyncSession) {
     this._internal = internal;
+    SyncSession.instances.add(new binding.WeakRef(this));
```

Comment:
Won't this grow forever in the non-test case? I don't see anything that removes dead WeakRefs from `instances` outside of tests.

Reply:
Good catch. I could add the removal to `resetInternal`? That way the closing a Realm would at least remove it.
If we had the `FinalizationRegistry` available on all platforms, that might also be used, but I'm not completely sure that we do.